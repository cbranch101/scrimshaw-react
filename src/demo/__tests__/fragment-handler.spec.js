import createFragmentHandler from "../fragment-handler.js"

test("should replace any defined fragments in a query", () => {
    const fragmentHandler = createFragmentHandler()
    const fragmentOne = fragmentHandler.createFragment(
        `
        fragment on Test {
            name
        }
    `,
        "fragmentName"
    )
    const query = `
        {
            test {
                ...${fragmentOne}
            }
        }
    `
    const wrappedQuery = fragmentHandler.mixFragmentsIntoQuery(query)
    expect(wrappedQuery).toMatchSnapshot()
})
