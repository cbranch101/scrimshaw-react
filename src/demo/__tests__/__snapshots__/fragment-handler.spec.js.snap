// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should replace any defined fragments in a query 1`] = `
"
        {
            test {
                ...fragmentName
            }
        }
    

        fragment fragmentName on Test {
            name
        }
    "
`;
